crates/pipeline_manager/src/db/mod.rs

```diff
@@ -246,87 +241,42 @@ pub(crate) struct ProjectDescr {
     pub schema: Option<String>,
 }
 
-/// Project configuration descriptor.
+/// PipelineDescr descriptor.
 #[derive(Serialize, ToSchema, Eq, PartialEq, Debug, Clone)]
-pub(crate) struct ConfigDescr {
-    pub config_id: ConfigId,
-    pub project_id: Option<ProjectId>,
-    pub pipeline: Option<PipelineDescr>,
+pub(crate) struct PipelineDescr {
+    pub pipeline_id: PipelineId,
+    pub program_id: Option<ProgramId>,
     pub version: Version,
     pub name: String,
     pub description: String,
     pub config: String,
     pub attached_connectors: Vec<AttachedConnector>,
+    pub port: u16,
+    pub shutdown: bool,
+    pub created: DateTime<Utc>,
 }
 
 /// Format to add attached connectors during a config update.
 #[derive(Deserialize, Serialize, ToSchema, Eq, PartialEq, Debug, Clone)]
 #[cfg_attr(test, derive(proptest_derive::Arbitrary))]
 pub(crate) struct AttachedConnector {
     /// A unique identifier for this attachement.
-    pub uuid: String,
+    pub name: String,
```

Comment:
Shall we switch to automatically generated uuids in the future? There doesn't seem to be a lot of values in having users come up with these names.

Reply:
So there is both now: we have server side generated uuids and client side generated names (string).

The pipeline, program and connector entities have name (unique, given by client) and uuid (unique, given by server)

The AttachedConnector doesn't have/need a uuid because it is never retrieved on it's own (it always comes 'attached' with a pipelinedescr).

Reply:
Fair enough. Still, does it need to have a name, and does the name have to be unique?

Reply:
We wanted to fix the ephemeral problem in the APIs by having a name, e.g., you can detect if something already exists by having a name that is unique. Note though that the attached connector name only needs to be unique within a pipeline (line 1201 in db/mod.rs). Whereas connector,program,pipeline names are unique for a user (right now entire table).

Having a name for attached connector is nice because I can identify the right node in the UI graph.

Reply:
I think it's important to have human-supplied names and server-side generated UUIDs. The names can be unique within a namespace (for now, full table, but soon, per user).